tests/testthat/test-lint_file.R

```diff
@@ -109,3 +109,16 @@ test_that("lint() results do not depend on the position of the .lintr", {
     )
   )
 })
+
+test_that("lint uses linter names", {
+  expect_lint("a = 2", list(linter = "bla"), linters = list(bla = assignment_linter), parse_settings = FALSE)
+})
+
+test_that("exclusions work with custom linter names", {
+  expect_lint(
+    "a = 2 # nolint: bla.",
```

Comment:
If we're going to allow this, I hope it will be pretty clearly documented -- I can imagine it being a bit hard to track down what's happening as an outside contributor to a package that takes advantage of this...

Reply:
I agree we should document all of #660 extremely clearly.
For configurable linters which might be used with more than one set of options (e.g. `undesirable_functions_linter`) I can't really see any alternative way that would allow a nicely formatted exclude tag.

Note that the auto names given by `with_defaults` are identical to the linter names, so most of the time this feature won't matter much.

Reply:
OK that makes sense. This is a good example to put in the docs/NEWS.